lib/OpenQA/Task/Job/ArchiveResults.pm

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Task::Job::ArchiveResults;
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
+
+sub register ($self, $app, @args) {
+    $app->minion->add_task(archive_job_results => \&_archive_results);
+}
+
+sub _archive_results ($minion_job, @args) {
+    my ($openqa_job_id) = @args;
+    my $app = $minion_job->app;
+    return $minion_job->fail('No job ID specified.') unless defined $openqa_job_id;
+
+    # FIXME: Should we ensure that the job results are finalized?
```

Comment:
how about we simply ignore non-finalized job results (can we detect that?) as they should be found as archiving candidates the next time

Reply:
> can we detect that?

That's a good question. I think we'd have to check whether there are any `.txt` files within the results dir.

> as they should be found as archiving candidates the next time

If finalizing a job fails for some reason it is not automatically re-attempted. So we should at least try to re-attempt the finalizing.